torchtitan/parallelisms/parallelize_llama.py

```diff
@@ -222,13 +389,17 @@ def parallelize_llama(model, world_mesh, parallel_dims, job_config: JobConfig):
                 )
             # As an optimization, do not reshard after forward for the last
             # transformer block since FSDP would prefetch it immediately
-            reshard_after_forward = int(layer_id) < len(model.layers) - 1
+            # reshard_after_forward = int(layer_id) < len(model.layers) - 1
+            # TODO(whc) need to fix correctly handle layer-ids on pp-split module
+            reshard_after_forward = True
```

Comment:
@awgu I'm confused, i think you mentioned i need to set this to true a while back.  In most of the PP schedules, i probably want to disable resharding so i get zero-2, don't I?

Why would i want this to be set to true?

Reply:
I might have misspoke. We want to set this to `False` for PP.